eZ/Publish/API/Repository/Tests/URLAliasServiceTest.php

```diff
@@ -1600,6 +1600,57 @@ public function testDeleteCorruptedUrlAliases()
         );
     }
 
+    public function testRenamingParentContentDoesntBreakChildAlias(): void
+    {
+        $repository = $this->getRepository();
+        $locationService = $repository->getLocationService();
+        $urlAliasService = $repository->getURLAliasService();
+        $contentTypeService = $repository->getContentTypeService();
+        $contentService = $repository->getContentService();
+
+        $languageCode = 'eng-GB';
+        $contentType = $contentTypeService->loadContentTypeByIdentifier('folder');
+
+        // 1. Create parent folder
+        $folderCreateStruct = $contentService->newContentCreateStruct($contentType, $languageCode);
+        $folderCreateStruct->setField('name', 'A');
+        $folderDraft = $contentService->createContent($folderCreateStruct, [
+            $locationService->newLocationCreateStruct(2),
+        ]);
+        $folder = $contentService->publishVersion($folderDraft->getVersionInfo());
+        $folderLocation = $locationService->loadLocation($folder->contentInfo->getMainLocationId());
+
+        // 2. Create child folder
+        $childCreateStruct = $contentService->newContentCreateStruct($contentType, $languageCode);
+        $childCreateStruct->setField('title', 'B');
+        $childDraft = $contentService->createContent($folderCreateStruct, [
+            $locationService->newLocationCreateStruct($folderLocation->id),
+        ]);
+        $child = $contentService->publishVersion($childDraft->getVersionInfo());
+        $childLocation = $locationService->loadLocation($child->contentInfo->getMainLocationId());
+
+        // 3. Create custom URL alias for child folder
+        $urlAliasService->createUrlAlias($childLocation, '/c/b', $languageCode);
+        $lookup = $urlAliasService->lookup('/c/b');
+
+        self::assertSame('/c/b', $lookup->path);
+
+        // 4. Rename "A" to "C"
+        $folderDraft = $contentService->createContentDraft($folder->contentInfo);
+        $folderUpdateStruct = $contentService->newContentUpdateStruct();
+        $folderUpdateStruct->setField('name', 'C');
+        $renamedFolder = $contentService->updateContent($folderDraft->getVersionInfo(), $folderUpdateStruct);
+        $contentService->publishVersion($renamedFolder->getVersionInfo());
+
+        // Loading aliases shouldn't throw a `BadStateException`
+        $urlAliasService->listLocationAliases($childLocation);
```

Comment:
can we actually check if the list is as expected? Not sure how big that snippet would be and if we're not gonna run into (Postgre)SQL returned records issues. But still sth to consider.